apps/web/src/app/api/evaluations/route.ts

```diff
@@ -17,22 +18,25 @@ export const GET = errorHandler(
       const searchParams = request.nextUrl.searchParams
       const {
         projectId: projectIdParam,
-        commitUuid,
+        commitUuid: commitUuidParam,
         documentUuid,
       } = searchParamsSchema.parse({
         projectId: searchParams.get('projectId') ?? undefined,
         commitUuid: searchParams.get('commitUuid') ?? undefined,
         documentUuid: searchParams.get('documentUuid') ?? undefined,
       })
 
-      const evaluationsRepository = new EvaluationsV2Repository(workspace.id)
-      const evaluations = await evaluationsRepository
-        .list({
-          projectId: projectIdParam ? Number(projectIdParam) : undefined,
-          commitUuid,
-          documentUuid,
-        })
-        .then((r) => r.unwrap())
+      const projectId = projectIdParam ? Number(projectIdParam) : undefined
+      const commitUuid = commitUuidParam || HEAD_COMMIT
+
+      const repository = new EvaluationsV2Repository(workspace.id)
+      const evaluations = documentUuid
+        ? await repository
+            .listAtCommitByDocument({ projectId, commitUuid, documentUuid })
+            .then((r) => r.unwrap())
+        : await repository
+            .listAtCommit({ projectId, commitUuid })
+            .then((r) => r.unwrap())
```

Comment:
hmm the repository is not good enough for composing filters. If I want another filter i will need to create another repo method + add another if statement here. We will have to figure out something better in the future.